source/ContentScript/recorder.ts

```diff
@@ -86,14 +97,18 @@ class Recorder {
     }
     if (this.curLevel > level) {
       while (this.curLevel > level) {
-        this.sendZestScriptToZAP(new ZestStatementSwitchToFrame(-1));
+        this.sendZestScriptToZAP(
+          new ZestStatementSwitchToFrame(-1, this.getWaited())
```

Comment:
ZestStatementSwitchToFrame doesnt currently support the waitForMsec param. 
That was only added to the [ZestClientElement](https://github.com/zaproxy/zest/blob/main/src/main/java/org/zaproxy/zest/core/v1/ZestClientElement.java) class.